Mage/src/main/java/mage/abilities/effects/common/combat/CanBlockAdditionalCreatureAllEffect.java

```diff
@@ -79,7 +67,29 @@ private String setText() {
     }
 
     @Override
-    public boolean hasLayer(Layer layer) {
-        return layer == Layer.RulesEffects;
+    public boolean applyToObjects(Layer layer, SubLayer sublayer, Ability source, Game game, List<MageItem> objects) {
+        for (MageItem object : objects) {
+            if (!(object instanceof Permanent)) {
+                continue;
+            }
+            Permanent permanent = (Permanent) object;
+            if (amount > 0) {
+                if (permanent.getMaxBlocks() > 0) {
+                    permanent.setMaxBlocks(permanent.getMaxBlocks() + amount);
+                }
+            } else {
+                permanent.setMaxBlocks(0);
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public List<MageItem> queryAffectedObjects(Layer layer, Ability source, Game game) {
+        return game.getBattlefield()
+                .getActivePermanents(filter, source.getControllerId(), source, game)
+                .stream()
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
```

Comment:
1. Why you use 2 methods (query and applyToObjects) instead single method?
2. Create objects list in parent code like EffectsList and put as param instead create whole list every call in every method. It’s cleaner (affected objects list used by single effect anyway).
3. Do not modify whole effects. Try few effects and ask about review and feedback.